pkg/scaffold/render.go

```diff
@@ -0,0 +1,24 @@
+// Copyright 2018 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package scaffold
+
+import "io"
+
+// Codegen is the interface for code generation.
+// Each to be generated file should have an implementation that implements this interface.
+type Codegen interface {
```

Comment:
Minor nit but `Codegen` is too general and overloaded in my opinion. 
Plus it's used in other places like codegen for CR.

How about something more specific like `TemplateGenerator` or `Scaffold`? Although `Scaffold` is still a bit general.
We don't need to block on this though. We can rename this later.

Reply:
yeah, I agree. it is really a place holder for something better. Let's revisit this later.